### DISCUSSION TITLE
Add custom javascript to code-server

### DISCUSSION THREAD

**Discussion by sfiggins** [coder/code-server#5506] [Q&A] [answered]
I would like to add a bit of javascript to the start up of code-server to capture some cross window communication and invoke some commands in a vs-code extension.

In version 3.x we altered the `src/browser/pages/vscode.html` in a prebuilt release so it required our code. This looks like it might be more difficult in 4.x.

Similar to [a recently answered question about customizing the UI](https://github.com/coder/code-server/discussions/5499), one way to do this might be to patch `lib/vscode/out/vs/code/browser/workbench/workbench-dev.html` and then build our own release, but is there an easier way to inject a bit of code into the code-server's window without rebuilding?

**Comment by jsjoeio** [comment, marked as answer]
> workbench-dev.html

this is only used in development so i believe you'd need to modify `workbench.html` instead. 

> but is there an easier way to inject a bit of code into the code-server's window without rebuilding

i believe that's the only way I know. maybe a browser extension?

**Reply by sfiggins** [reply]
Updating the right file helped out. And we didn't need to rebuild vs code to get it to work, just updated it in a pre built distribution. Thanks!
